packages/core/src/crypto/JwkTypes.ts

```diff
@@ -1,6 +1,157 @@
+import { KeyType } from './KeyType'
+
 export interface Jwk {
   kty: 'EC' | 'OKP'
-  crv: 'Ed25519' | 'X25519' | 'P-256' | 'P-384' | 'secp256k1'
+  crv: 'Ed25519' | 'X25519' | 'P-256' | 'P-384' | 'P-521' | 'Bls12381G1' | 'Bls12381G2'
   x: string
   y?: string
+  use?: 'sig' | 'enc'
+}
+
+export interface Ed25519Jwk {
+  kty: 'OKP'
+  crv: 'Ed25519'
+  x: string
+  use?: 'sig'
+}
+
+export interface Bls12381g1Jwk {
+  kty: 'EC'
+  crv: 'Bls12381G1'
+  x: string
+  use?: 'sig'
+}
+
+export interface Bls12381g2Jwk {
+  kty: 'EC'
+  crv: 'Bls12381G2'
+  x: string
+  use?: 'sig'
+}
+
+export interface X25519Jwk {
+  kty: 'OKP'
+  crv: 'X25519'
+  x: string
+  use?: 'enc'
+}
+
+export interface P256Jwk {
+  kty: 'EC'
+  crv: 'P-256'
+  x: string
+  y: string
+  use?: 'sig' | 'enc'
+}
+
+export interface P384Jwk {
+  kty: 'EC'
+  crv: 'P-384'
+  x: string
+  y: string
+  use?: 'sig' | 'enc'
+}
+
+export interface P521Jwk {
+  kty: 'EC'
+  crv: 'P-521'
+  x: string
+  y: string
+  use?: 'sig' | 'enc'
+}
+
+export function isEd25519Jwk(jwk: Jwk): jwk is Ed25519Jwk {
+  return jwk.kty === 'OKP' && jwk.crv === 'Ed25519' && jwk.x !== undefined && (!jwk.use || jwk.use === 'sig')
+}
+
+export function isX25519Jwk(jwk: Jwk): jwk is X25519Jwk {
```

Comment:
Do we want to add the `x` check here? Without the X it can still be a valid x25519, just not one that contains a public key. Maybe a rename if that behaviour is desired?

Reply:
Good point. So you mean call it `isX25519PublicKeyJwk`?